### ISSUE TITLE
Undefined variables in function calls are not detected

### ISSUE THREAD

**Issue by GaryJones** [sirbrillig/phpcs-variable-analysis#233]
Snippet:

```php
<?php

if ( ! empty( $this->vars['video-id'] ) ) {
	echo $this->vars['video-id'];
}
```

With a default configuration (`phpcs --standard=VariableAnalysis -s phpcs-test.php`), then "Variable $this is undefined. (VariableAnalysis.CodeAnalysis.VariableAnalysis.UndefinedVariable)" is reported for line 4 (the `echo` line), but not for the `$this` in the `empty()`.

Likewise: 

```php
<?php

if ( empty( $this->vars['video-id'] ) ) {
	return;
}
```

...doesn't give any violation at all.

If I remove the `empty()`, then the violation is reported.

```php
<?php

if ( ! $this->vars['video-id'] ) {
	return;
}
```

Tested in VA 2.11.0.

**Comment by sirbrillig** [comment]
If the only function you tested with is `empty()`, then this is not a bug but a feature since `empty()` is allowed to contain undefined variables (since https://github.com/sirbrillig/phpcs-variable-analysis/pull/204). This feature was added because both `empty()` and `isset()` are language constructs that are often used to detect the existence of variables (for example, no warning or error is thrown doing `empty( $foo['bar']['baz'] )` if `$foo` does not exist, but will throw warnings for any other function like `do_something( $foo['bar']['baz'] )`). 

Do you find this occurs with other functions as well?

**Comment by GaryJones** [comment]
That makes sense. I tried with `current_user_can()` as the wrapping function, and that correctly gave a violation, and then this was correctly silenced with the `<property name="allowUndefinedVariablesInFileScope" value="true"/>` line in the config file ✅
